src/components/LogViewer/LogViewerToolbar.tsx

```diff
@@ -24,74 +26,120 @@ export const LogViewerToolbar = memo(
-                logViewerRef.current?.scrollToBottom();
-                setIsPaused(false);
-              }}
-              variant="control"
-              icon={<LongArrowAltDownIcon />}
-            >
-              Bottom
-            </Button>
-          </ToolbarItem>
-        </ToolbarGroup>
-        <ToolbarGroup align={{ default: 'alignRight' }}>
-          {!isStatic && (
-            <>
-              <ToolbarItem alignSelf="center">
-                <TooltipWrapper tooltip="Scroll down automatically on new log lines input.">
-                  <Checkbox
-                    id="force-following-check"
-                    label="Force following"
-                    isChecked={isFollowing}
-                    onChange={(_, checked) => {
-                      setIsPaused(!checked);
-                      setIsFollowing(checked);
-                    }}
-                  />
-                </TooltipWrapper>
-              </ToolbarItem>
-              <ToolbarItem variant="separator" />
-            </>
-          )}
-          <ToolbarItem alignSelf="center">
-            <Checkbox
-              id="wrap-lines-check"
-              label="Wrap lines"
-              isChecked={areLinesWrapped}
-              onChange={(_, checked) => {
-                setAreLinesWrapped(checked);
-              }}
-            />
-          </ToolbarItem>
-          {!!customActions?.length &&
-            customActions.map((node, index) => (
-              <Fragment key={index}>
+  }: ILogViewerToolbarProps) => {
+    useEffect(() => {
+      const handleKeyDown = (event: KeyboardEvent) => {
+        const target = event.target as HTMLElement;
+        const tagName = target.tagName.toLowerCase();
+        const isInput = tagName === 'input' || tagName === 'textarea' || target.isContentEditable;
+
+        if (isInput) {
+          return;
+        }
+
+        if (event.key === '/') {
+          event.preventDefault();
+          const searchInput = document.querySelector<HTMLInputElement>(
+            '.pf-v5-c-log-viewer__header .pf-v5-c-text-input-group__text-input'
+          );
+          searchInput?.focus();
+        }
+      };
+
+      window.addEventListener('keydown', handleKeyDown);
+      return () => {
+        window.removeEventListener('keydown', handleKeyDown);
+      };
+    }, []);
+
+    useEffect(() => {
+      if (autofocusSearchBar) {
+        const timeoutId = setTimeout(() => {
+          const searchInput = document.querySelector<HTMLInputElement>(
+            '.pf-v5-c-log-viewer__header .pf-v5-c-text-input-group__text-input'
+          );
+          searchInput?.focus();
+        }, 0);
+
+        return () => clearTimeout(timeoutId);
+      }
+    }, [autofocusSearchBar]);
+
+    return (
+      <Toolbar>
+        <ToolbarContent alignItems="center">
+          <ToolbarGroup>
+            <ToolbarItem>
+              <LogViewerSearch minSearchChars={1} placeholder="Type / to search" />
```

Comment:
Ideally separate the search bar and all related logic outside from the toolbar

Reply:
Just like Top and Bottom buttons, the search bar is highly bonded to the toolbar of the log viewer only, and search bar will need to be used for every log viewer(as Martin mentioned earlier in this PR), so I think we should keep it in toolbar.

Reply:
@patrikk0123 updated